presto-native-execution/presto_cpp/main/types/PrestoToVeloxExpr.cpp

```diff
@@ -159,13 +159,48 @@ std::vector<TypedExprPtr> VeloxExprConverter::toVeloxExpr(
 }
 
 namespace {
+static const char* kVarchar = "varchar";
+
+/// Convert cast to a call to substr if target type is varchar with max length.
+/// Convert try_cast to a call to try(substr) if target type is varchar with max
```

Comment:
@gggrace14 try_cast is not equivalent to try(substr). See https://velox-lib.io/blog/#try-and-trycast